### PR TITLE
chore: fix nodejs sample app

### PR DESCRIPTION
**Pull request by saikumarrs**
I've added Node.js sample apps for JavaScript and TypeScript languages.
See the README.md files in the `examples/<app>` directory for more details.

Note that the sample apps are now added in the root `examples` directory instead of `tests/e2e`.

I've also added steps to run the sample app against regenerated client libraries from both the production version of RudderTyper and the local working copy in the repository.
This will help us during development where we have to validate the changes quickly.

Apart from that the following major changes were made:
- I replaced `yarn` with `npm` throughout the repo (we have to update the docs if needed).
- Fixed GitHub workflows and added CODEOWNERS file.
- Upgrade Node to the latest v20.x.x version.
- Added `husky` dependency for pre-commit hooks.
- Removed redundant e2e test setup as it was not applicable anymore.
- Updated README.md in the root to the latest.
- Upgraded all the dependencies to their latest versions and used absolute versions.

These changes will now enable us to run RudderTyper tool in any terminal client.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+    <img alt="RudderStack" width="512" src="https://raw.githubusercontent.com/rudderlabs/rudder-typer/master/assets/rs-logo-full-light.jpg">
+  </a>
+  <br />
+  <caption>The Customer Data Platform for Developers</caption>
+</p>
+<p align="center">
+  <b>
+    <a href="https://rudderstack.com">Website</a>
+    ·
+    <a href="https://rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/">Documentation</a>
```
Comment (akashrpo) [inline README.md:12]:
```suggestion
    <a href="https://www.rudderstack.com/docs/data-governance/tracking-plans/ruddertyper/">Documentation</a>
```

File: src/cli/commands/build.tsx [hunk 2]
```diff
@@ -202,7 +202,7 @@ export const UpdatePlanStep: React.FC<UpdatePlanStepProps> = ({
         id: toTrackingPlanId(newTrackingPlan),
         version: newTrackingPlan.version,
         path: trackingPlanConfig.path,
-        trackCalls: events.map<JSONSchema7>(e => ({
+        trackCalls: events.map<JSONSchema7>((e) => ({
```
Comment (akashrpo) [inline src/cli/commands/build.tsx:205]: Is it recommended to add brackets here? When my prettier auto prettifies, it removes this bracket and in the rest of the files

File: src/cli/commands/init.tsx [hunk 3]
```diff
@@ -156,16 +156,17 @@ export const Init: React.FC<InitProps> = props => {
 const Header: React.FC = () => {
   return (
     <Box flexDirection="column">
-      <Box width={80} textWrap="wrap" marginBottom={4}>
-        <Color white>
-          RudderTyper is a tool for generating strongly-typed{' '}
+      <Box width={80} marginBottom={4}>
```
Comment (akashrpo) [inline src/cli/commands/init.tsx:159]:
```suggestion
      <Box width={80} marginBottom={4} flexDirection="column">
```

File: src/cli/commands/build.tsx (hunk 2 above)
Comment (saikumarrs) [reply src/cli/commands/build.tsx:205]: You can point your prettier extensions to `.prettierrc` file of the project to eliminate inconsistencies.

File: src/cli/commands/init.tsx (hunk 3 above)
Comment (saikumarrs) [reply src/cli/commands/init.tsx:159]: Does this fix the UI issues you've been observing?

File: src/cli/commands/init.tsx [hunk 4]
```diff
@@ -559,7 +620,8 @@ const APITokenPrompt: React.FC<APITokenPromptProps> = ({ step, config, configPat
         {state.canBeSet && !state.foundCachedToken && (
           <Box flexDirection="column">
             <Box>
-              <Text>{figures.pointer}</Text>{' '}
+              <Text>{figures.pointer}</Text>
+              <Text> </Text>
               <TextInput
                 value={state.token}
```
Comment (akashrpo) [inline src/cli/commands/init.tsx:626]: We have unused return here

File: src/cli/commands/init.tsx [hunk 5]
```diff
@@ -529,12 +531,71 @@ const APITokenPrompt: React.FC<APITokenPromptProps> = ({ step, config, configPat
...
+                onSubmit={onConfirm}
+              />
+            </Box>
+          </Box>
+        </Step>
+      )}
+    </>
+  );
+
   return (
```
Comment (akashrpo) [inline src/cli/commands/init.tsx:599]: We have unused return here

Comment (saikumarrs) [reply src/cli/commands/init.tsx]: Done.

File: src/cli/commands/init.tsx [hunk 6]
```diff
@@ -199,23 +207,23 @@ type SDKPromptProps = {
...
   };
 
   const tips = [
-    'Use your arrow keys to select.',
-    'RudderTyper clients are strongly-typed wrappers around a RudderStack SDK.',
+    'Use the arrow keys to navigate.',
+    'RudderTyper clients are strongly-typed wrappers for the RudderStack SDK.',
     <Text key="sdk-docs">
-      To learn more about {"RudderStack's"} SDKs, see the{' '}
+      To learn more about the {'RudderStack'} SDKs, visit the
```
Comment (akashrpo) [inline src/cli/commands/init.tsx:226]:
```suggestion
      To learn more about the {'RudderStack'} SDKs, visit the<Text> </Text>
```

File: src/cli/commands/init.tsx [hunk 7]
```diff
@@ -517,9 +532,9 @@ const APITokenPrompt: React.FC<APITokenPromptProps> = ({ step, config, configPat
   };
 
   const tips = [
-    'An API token is used to download Tracking Plans from Rudder.',
+    'An API token is required to download Tracking Plans from RudderStack.',
     <Text key="api-token-docs">
-      Documentation on generating an API token can be found{' '}
+      You can find documentation on generating an API token
```
Comment (akashrpo) [inline src/cli/commands/init.tsx:537]:
```suggestion
      You can find documentation on generating an API token<Text> </Text>
```

File: README.md [hunk 8]
```diff
@@ -180,14 +200,19 @@ import com.rudderstack.generated.*
 
 ```javascript
 // Import RudderStack JS SDK and initialize it
-const rudderanalytics = require('rudder-sdk-js');
-rudderanalytics.load(YOUR_WRITE_KEY, DATA_PLANE_URL);
+var RudderAnalytics = require('@rudderstack/analytics-js');
```
Comment (akashrpo) [inline README.md:203]:
```suggestion
const RudderAnalytics = require('@rudderstack/analytics-js');
```

File: README.md [hunk 9]
```diff
@@ -180,14 +200,19 @@ import com.rudderstack.generated.*
...
+
+const rudderAnalytics = new RudderAnalytics();
+rudderAnalytics.load(WRITE_KEY, DATA_PLANE_URL, {});
+
 // Import your auto-generated RudderTyper client:
-const rudderTyper = require('./rudderTyperClient');
-// Pass in your rudder-sdk-js instance to RudderTyper client
+const rudderTyper = require('./analytics/index');
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md:209]: I'll let @MoumitaM address this in her branch.

File: README.md (hunk 8 above)
Comment (saikumarrs) [reply README.md]: I'll let @MoumitaM address this in her branch.

File: README.md [hunk 10]
```diff
@@ -250,19 +275,25 @@ browserify rudderTyperClient.js --standalone rudderTyper >  rudderTyperBundle.js
...
-// Pass in your rudder-sdk-node instance to RudderTyper.
+// Import Rudder Node SDK and initialize it
+const RudderAnalytics = require('@rudderstack/rudder-sdk-node');
+
+const client = new RudderAnalytics(WRITE_KEY, {
+  dataPlaneUrl: DATA_PLANE_URL,
+  // More initialization options
+});
+
+const ruddertyper = require('./analytics/index');
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md:291]: Addressed it accordingly.

File: README.md [hunk 11]
```diff
@@ -223,8 +248,8 @@ browserify rudderTyperClient.js --standalone rudderTyper >  rudderTyperBundle.js
     ];
     for (var i = 0; i < methods.length; i++) {
       var method = methods[i];
-      rudderanalytics[method] = (function(methodName) {
-        return function() {
+      rudderanalytics[method] = (function (methodName) {
+        return function () {
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md:252]: I'll let @MoumitaM address this in her branch.

**Comment by akashrpo** [comment]
Do we have to update the [license](https://github.com/rudderlabs/rudder-typer/blob/develop/.github/LICENSE) ?

**Review by akashrpo — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Pull request merged**